### PR TITLE
feat(m3): harden run interruption escalation and capture drain

### LINKED ISSUE #79: Harden run interruption escalation and capture join edge cases
## Context
Gemini feedback on PR #73 points to remaining resilience gaps in runtime interruption/capture behavior.

Source: https://github.com/manuelpenazuniga/ClawCrate/pull/73

## Problem
Current monitoring loop:
- treats first interrupt gracefully, but subsequent interrupts are ignored until grace-path logic triggers.
- can still block on capture thread join if descendants keep inherited stdio pipes open.

## Risk
`clawcrate run` can feel unresponsive under repeated Ctrl-C and may hang after child exit in edge process-tree cases.

## Proposed Fix
1. Support immediate escalation on repeated SIGINT/SIGTERM.
2. Harden capture shutdown/join behavior to avoid indefinite wait when pipe inheritance keeps streams open.
3. Preserve final artifact integrity guarantees.

## Done When
- Repeated interrupt reliably forces fast termination.
- Capture lifecycle avoids indefinite hang in inherited-pipe scenarios.
- Tests cover interruption escalation and pipe-inheritance edge case.

Related: #4

### PR DESCRIPTION
**Pull request by manuelpenazuniga**
## Summary

- escalate run interruption immediately on repeated SIGINT/SIGTERM instead of waiting full grace period
- place sandboxed child processes in dedicated process groups and signal the group on termination/kill paths
- make capture drain fail-safe with bounded wait + forced cleanup to avoid hangs when descendants keep inherited stdio open
- add deterministic unit tests for interruption escalation and inherited-pipe descendant cleanup

## Validation
- cargo fmt --all
- cargo test -p clawcrate-cli
- cargo clippy -p clawcrate-cli --all-targets -- -D warnings
- cargo test --workspace
- cargo clippy --workspace --all-targets -- -D warnings

Closes #79

### CODE REVIEW

File: crates/clawcrate-cli/src/main.rs [hunk 1]
```diff
@@ -1589,37 +1657,55 @@ fn run_monitored_child(
 }
 
 #[cfg(unix)]
-fn send_termination_signal(child: &Child) -> Result<()> {
-    send_unix_signal(child, Signal::SIGTERM)
+fn send_termination_signal(child: &Child, process_group_id: Option<i32>) -> Result<()> {
+    send_unix_signal_to_pid(child.id() as i32, Signal::SIGTERM)?;
+    if let Some(group_id) = process_group_id.filter(|group_id| *group_id > 0) {
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-cli/src/main.rs:1662]: @manuelpenazuniga check that!

File: crates/clawcrate-cli/src/main.rs [hunk 2]
```diff
@@ -1589,37 +1657,55 @@ fn run_monitored_child(
...
         .kill()
         .map_err(|source| anyhow!("failed to terminate child process: {source}"))
 }
 
 #[cfg(unix)]
-fn send_kill_signal(child: &Child) -> Result<()> {
-    send_unix_signal(child, Signal::SIGKILL)
+fn send_kill_signal(child: &Child, process_group_id: Option<i32>) -> Result<()> {
+    send_unix_signal_to_pid(child.id() as i32, Signal::SIGKILL)?;
+    if let Some(group_id) = process_group_id.filter(|group_id| *group_id > 0) {
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-cli/src/main.rs:1678]: @manuelpenazuniga review!

File: crates/clawcrate-cli/src/main.rs [hunk 3]
```diff
@@ -1567,10 +1613,32 @@ fn run_monitored_child(
...
+                {
+                    send_kill_signal(child, process_group_id)?;
+                    capture_force_kill_sent = true;
+                }
+
+                if capture_wait_started.elapsed() >= CAPTURE_DRAIN_MAX_WAIT {
+                    return Err(anyhow!(
+                        "timed out draining captured output; descendant process may be holding stdout/stderr open"
+                    ));
+                }
```
[the remark this answers is not in the document]
Comment (manuelpenazuniga) [reply crates/clawcrate-cli/src/main.rs:1639]: okok
